platform/kube/controller.go

```diff
@@ -630,34 +630,37 @@ const (
 // An empty array is always returned if an error occurs.
 func (c *Controller) GetIstioServiceAccounts(hostname string) ([]string, error) {
 	name, namespace, err := parseHostname(hostname)
-	saArray := make([]string, 0)
 	if err != nil {
 		glog.Warningf("parseHostname(%s) => error %v", hostname, err)
-		return saArray, err
+		return nil, err
 	}
 	svc, exists := c.serviceByKey(name, namespace)
 	if !exists {
 		err := fmt.Sprintf("Failed to get service for hostname %s.", hostname)
 		glog.Warningf(err)
-		return saArray, errors.New(err)
-	}
-	lo := v1.ListOptions{
-		LabelSelector: labels.Set(svc.Spec.Selector).String(),
-	}
-	// TODO: This is fragile, improve it.
-	pods, err := c.client.client.CoreV1().Pods(svc.Namespace).List(lo)
-	if err != nil {
-		glog.Warningf("Failed to get pods for service %s.", hostname)
-		return saArray, err
+		return nil, errors.New(err)
```

Comment:
Actually, it would be better to let the caller handle/print the error here, otherwise we could end up with a double-printed error message in the logs in the future, e.g.

```golang
if !exists {
    return nil, fmt.Errorf("failed to get service for hostname %v", hostname)
}

// Make buildSSLContextWithSAN handle the error

serviceAccounts, err := context.Discovery. GetIstioServiceAccounts(hostname)
if err == nil  {
    return &SSLContextWithSAN{...}
} 
glog.Warningf("Error: %v", err)
```